File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
class Solution {
    public int[] solution(int[] sequence, int k) {
        int[] answer = new int[2];
        int s = 0, e = 0, sum = 0;
        int min = 1000000001;
        int len = sequence.length;
        while(s <= e) {
            if(sum < k) {
                if(e >= len) break;
                sum += sequence[e];
                e++;
            }
            else if (sum > k) {
                sum -= sequence[s];
                s++;
            }
            else {
                if (e - 1 - s < min) {
                    min = e - 1 - s;
                    answer[0] = s;
                    answer[1] = e - 1;
                }
                sum -= sequence[s];
                s++;
            }
            
            
        }
        System.out.print(s + " " + e);
        return answer;
    }
}

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.js
function solution(sequence, k) {
    var answer = [];
    let start = 0, end = 0, sum = 0;
    const arr = [];
    while (end <= sequence.length) {
        if (sum < k) {
            sum += sequence[end++];
        } else if (sum > k) {
            sum -= sequence[start++];
        } else {
            arr.push([start,end-1]);
            sum -= sequence[start++];
            
        }
    }
    arr.sort((a,b) => {
        const lenA = a[1] - a[0];
        const lenB = b[1] - b[0];
        if (lenA !== lenB) return lenA - lenB;
        return a[0] - b[0];
    })
    answer = arr[0];
    return answer;
}